### PR TITLE
PMM-3673 Add Management API.

### PR DESCRIPTION
**Pull request by DexterHD**
@AlekSi What do you think about `request` and `response` parameters? Is it ok?

### CODE REVIEW

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-291/

File: api/managementpb/management.proto [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import "github.com/mwitkow/go-proto-validators/validator.proto";
+import "google/api/annotations.proto";
+import "inventory/agents.proto";
+import "inventory/services.proto";
+
+// Add MySQL
+
+message AddMySQLRequest {
+  // Node identifier where this instance runs. Required.
```
Comment (AlekSi) [inline api/managementpb/management.proto]: It is not clean what is "instance" there.

File: api/managementpb/management.proto [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  string password = 6;
+  // MySQL username for getting performance data.
+  string qan_username = 7 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // MySQL password for getting performance data.
+  string qan_password = 8;
+}
```
Comment (AlekSi) [inline api/managementpb/management.proto]: I think there should be a way to add only one Agent, or maybe even zero Agents. That what we had for PMM 1.x:
```
pmm-client add -h
This command is used to add a monitoring service.

Usage:
  pmm-admin add [command]

Available Commands:
  […]
  mysql              Add complete monitoring for MySQL instance (linux and mysql metrics, queries).
  mysql:metrics      Add MySQL instance to metrics monitoring.
  mysql:queries      Add MySQL instance to Query Analytics.
[…]
```

(also note how it uses "monitoring service" term 😐)

So I propose to add two boolean flags for that.

File: api/managementpb/management.proto [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+message AddMySQLResponse {
+  inventory.MySQLService service = 1;
+  inventory.MySQLdExporter agent = 2;
```
Comment (AlekSi) [inline api/managementpb/management.proto]:
```suggestion
  inventory.MySQLdExporter mysqld_exporter = 2;
```

It is likely we will have more.

File: api/managementpb/management.proto [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  inventory.QANMySQLPerfSchemaAgent qan_agent = 3;
```
Comment (AlekSi) [inline api/managementpb/management.proto]:
```suggestion
  inventory.QANMySQLPerfSchemaAgent qan_mysql_perfschema = 3;
```

Again, we will have more.

File: api/managementpb/management.proto [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+// Management service provides public methods for Management API.
+service Management {
```
Comment (AlekSi) [inline api/managementpb/management.proto]: Maybe that should be `MySQL` service in `management` package?
```
/v1/management/MySQL/Add
/v1/management/PostgreSQL/Remove
```

File: api/managementpb/management.proto [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  // AddMySQL adds MySQL.
```
Comment (AlekSi) [inline api/managementpb/management.proto]: _That_ should be expanded to describe what it will do.

File: api/managementpb/management.proto [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+// Add MySQL
+
+message AddMySQLRequest {
+  // Node identifier where this instance runs. Required.
+  string node_id = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
```
Comment (AlekSi) [inline api/managementpb/management.proto]: What about `pmm_agent_id`?

File: api/managementpb/management.proto (hunk 2 above)
Comment (DexterHD) [reply api/managementpb/management.proto]: > also note how it uses "monitoring service" term

Is it comment from Peter about the same thing?
https://confluence.percona.com/display/PMM/Inventory+API?focusedCommentId=21038419#comment-21038419

Comment (AlekSi) [reply api/managementpb/management.proto]: Yes. See also https://confluence.percona.com/display/PMM/PMM+1.x%3A+Managing+external+exporters+with+pmm-managed+and+pmm-admin and https://confluence.percona.com/display/PMM/External+exporters+improvements.
But in the end, we stick with our current definitions for Service and Agent. So let's say something along the lines:
> This API adds MySQL Service and starts several Agents.

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-291/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-291/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-291/

**Comment by DexterHD** [comment]
@AlekSi Fixed... Agrhhh... Sorry for `:` in commits. It's a habit. I'll try to make a note in front of the monitor 😄

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-291/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-291/

**Pull request merged**